tests/test_vmec.py

```diff
@@ -907,3 +910,449 @@ def test_vmec_boundary_subspace(DummyStellarator):
+                    else ptolemy_identity_fwd(xm, xn, quant, np.zeros_like(quant))
+                )
+                # negate nu since we save it as negative of what our convention is
+                # to comply with booz_xform notation
+                quant_mn = -quant_mn if name == "nu" else quant_mn
+                quant_mn = (
+                    quant_mn * data["psi_r"][0] if name == "sqrt(g)_B" else quant_mn
+                )
+
+                basis = DoubleFourierSeries(
+                    np.max(m), round(np.max(n) / eq.NFP), sym=False, NFP=eq.NFP
+                )
+                quant_mn = np.where(basis.modes[:, 1] < 0, -quant_mn, quant_mn)
+                quant_mn_desc_basis = np.zeros((basis.num_modes,))
+                for i, (mm, nn) in enumerate(zip(m, n / eq.NFP)):
+                    idx = basis.get_idx(L=0, M=mm, N=nn)
+                    quant_mn_desc_basis[idx] = quant_mn[surf_index, i]
+
+                quant_trans = Transform(grid=grid_boozer, basis=basis)
+                if np.any(quant_from_booz):  # add to it if it exists
+                    quant_from_booz += quant_trans.transform(quant_mn_desc_basis)
+                else:
+                    quant_from_booz = quant_trans.transform(quant_mn_desc_basis)
+
+            np.testing.assert_allclose(
+                quant_from_booz,
+                data[name],
+                atol=3e-5,
+                rtol=3e-5,
+                err_msg=f"{name} at surf index {surf_index}",
+            )
+
+
+@pytest.mark.unit
+def test_make_boozmn_output_against_hidden_symmetries_booz_xform(TmpDir):
+    """Test that booz_xform-style outputs compare well against C++ implementation."""
+    # testing against https://github.com/hiddenSymmetries/booz_xform/tree/main
+    # commit 881907058ece03
+    # load in HELIOTRON equilibrium
+    eq = get("HELIOTRON")
+    output_path = str(TmpDir.join("boozmn_out.nc"))
+
+    boozer_res = 20
+    # compare against a 100 surface Mboz=Nboz=20 run of HELIOTRON
+    # with the hidden symmetries C++ booz_xform implementation
+    # (ran on a wout created with VMECIO.save of HELIOTRON example with 100 surfs)
+    surfs = 100
+    Cpp_booz_output_path = (
+        f"./tests/inputs/boozmn_{surfs}_surfs_heliotron_sims_booz_{boozer_res}.nc"
+    )
+
+    # Use DESC to calculate the boozer harmonics and create a booz_xform style .nc file
+    make_boozmn_output(
+        eq, output_path, surfs=surfs, verbose=0, M_booz=boozer_res, N_booz=boozer_res
+    )
+    # load in the .nc file
+    file = Dataset(output_path, mode="r")
+
+    R_mnc = file.variables["rmnc_b"][:].filled()
+    Z_mns = file.variables["zmns_b"][:].filled()
+    B_mnc = file.variables["bmnc_b"][:].filled()
+    nu_mns = file.variables["pmns_b"][:].filled()
+    g_mnc = file.variables["gmn_b"][:].filled()
+
+    quantities = [R_mnc, Z_mns, B_mnc, g_mnc, nu_mns]
+    quant_names = ["R", "Z", "|B|", "sqrt(g)_B", "nu"]
+    quant_atols = [1e-4, 7e-5, 9e-6, 5e-3, 2e-7]
```

Comment:
Are you sure these atols are good choices? rtols would be preferable if you can use them. It's not obvious to me what order of magnitude the output coefficients should be, and if the Fourier coefficients are decaying exponentially these atols are probably just ignoring all the higher-order modes.

Reply:
I am not 100% sure these are great choices to be honest, but htese are the tolerances that I get when comparing. I could try to compare with more surfs (as the main issue with the comparison I think is that the booz xform uses finite diffs to interpolate to the half grid, so they have some inherent error introduced there which DESC won't have)

Reply:
It feels like there is some circular logic going on here: You think your new boozmn.nc output is correct because it passes your comparison tests, but here you admit that you set the tolerances to whatever they needed to be to make them pass. 

These tolerances are low enough that the code is _probably_ correct, but it depends on the scale of the field in the example you are testing with. And by using an absolute tolerance, I still suspect you are just ignoring most of the high-order Fourier modes. Can we try switching to relative tolerances? That would avoid those two issues.